### PR TITLE
Mg password

### PR DESCRIPTION
**Pull request by Gallup93**
Refactor password edit route to password controller

Add password_controller

Add password edit view page

Add password edit action

Refactor each block into ActiveRecord

### CODE REVIEW

**Comment by Gallup93** [comment]
Light at the end of the tunnel!

Light at the end of the tunnel!

Edit* - this is the inside of a paper towel roll

File: app/controllers/admin/merchants_controller.rb [hunk 1]
```diff
@@ -11,12 +11,12 @@ def show
...
     @merchant = Merchant.find(params[:id])
     if @merchant.active? == true
-      @merchant.items.each {|item| item.update_attribute(:active?, false)}
+      Item.where(merchant_id: params[:id]).update(active?: false)
       @merchant.update_attribute(:active?, false)
       redirect_to "/admin/merchants"
       flash[:notice] = "disabled #{@merchant.name}"
     else
-      @merchant.items.each {|item| item.update_attribute(:active?, true)}
+      Item.where(merchant_id: params[:id]).update(active?: true)
```
Comment (Gallup93) [inline app/controllers/admin/merchants_controller.rb:19]: basically an AR pro now. nbd

File: config/routes.rb [hunk 2]
```diff
@@ -80,7 +80,7 @@
   post "/profile", to: "users#update"
 
 # add to a user password controller
-  get "/password/edit", to: "users#edit_pass"
+  get "/password/edit", to: "password#edit"
```
[the remark this answers is not in the document]
Comment (stellakunzang) [reply config/routes.rb:75]: professional RESTler

File: config/routes.rb [hunk 3]
```diff
@@ -80,7 +80,7 @@
...
   post "/password", to: "users#update_pass"
```
Comment (stellakunzang) [inline config/routes.rb]: Any plan to add this one to the new password controller, too?

File: app/controllers/admin/merchants_controller.rb (hunk 1 above)
Comment (muydanny) [reply app/controllers/admin/merchants_controller.rb:19]: upvote

File: config/routes.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (muydanny) [reply config/routes.rb:75]: zzzZZZ

**Review by muydanny — APPROVED** [review]

**Pull request merged**